x/oracle/keeper/msg_server.go

```diff
@@ -161,6 +162,78 @@ func (ms msgServer) GovUpdateParams(
 	return &types.MsgGovUpdateParamsResponse{}, nil
 }
 
+// GovAddDenoms adds new assets to the AcceptList, and adds
+// it to the MandatoryList if specified.
+func (ms msgServer) GovAddDenoms(
+	goCtx context.Context,
+	msg *types.MsgGovAddDenoms,
+) (*types.MsgGovAddDenomsResponse, error) {
+	if msg.Authority != ms.authority {
+		err := errors.Wrapf(
+			types.ErrNoGovAuthority,
+			"invalid authority; expected %s, got %s",
+			ms.authority,
+			msg.Authority,
+		)
+		return nil, err
+	}
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	params := ms.GetParams(ctx)
+
+	plan := types.ParamUpdatePlan{
+		Keys:    []string{},
+		Height:  msg.Height,
+		Changes: params,
+	}
+	for _, denom := range msg.DenomList {
+		// if the AcceptList already contains this denom, and we're not
+		// adding it to the "mandatory" list, error out.
+		if plan.Changes.AcceptList.Contains(denom.SymbolDenom) && !msg.Mandatory {
+			err := errors.Wrapf(
+				types.ErrInvalidParamValue,
+				"denom already exists in acceptList: %s",
+				denom.SymbolDenom,
+			)
+			return nil, err
+			// if the MandatoryList already contains this denom, and we're trying to
+			// add it to the "mandatory" list, error out.
+		} else if plan.Changes.MandatoryList.Contains(denom.SymbolDenom) && msg.Mandatory {
+			err := errors.Wrapf(
+				types.ErrInvalidParamValue,
+				"denom already exists in mandatoryList: %s",
+				denom.SymbolDenom,
+			)
+			return nil, err
+		}
+
+		// add to AcceptList & MandatoryList if necessary
+		if !plan.Changes.AcceptList.Contains(denom.SymbolDenom) {
+			plan.Changes.AcceptList = append(plan.Changes.AcceptList, denom)
+			plan.Keys = ojoutils.AppendUniqueString(plan.Keys, string(types.KeyAcceptList))
+		}
+		if msg.Mandatory {
+			plan.Changes.MandatoryList = append(plan.Changes.MandatoryList, denom)
+			plan.Keys = ojoutils.AppendUniqueString(plan.Keys, string(types.KeyMandatoryList))
+		}
+
+		// add a RewardBand
+		_, err := plan.Changes.RewardBands.GetBandFromDenom(denom.SymbolDenom)
+		if err != nil {
+			plan.Changes.RewardBands.AddDefault(denom.SymbolDenom)
```

Comment:
How often are we going to use the default when adding new assets? Should we just require setting it in the tx?

Reply:
🤷 I could go either way on that
We could add an option and if it's not provided we can just default it

Reply:
I like the idea of adding it. Either make it required or use the default when not set.

Reply:
Added the option [here](https://github.com/ojo-network/ojo/pull/283/commits/669e9eae6a124c11f087621eccc0d7e20a79bc60)